src/Repositories/Eloquent/ProductImageRepository.php

```diff
@@ -76,18 +76,19 @@ public function delete(int $id): bool
     {
         $this->db->beginTransaction();
 
-        $model = $this->model()->find($id);
+        try {
+            $model = $this->findById($id);
 
-        if ($model) {
             if ($model->delete()) {
-                if ($this->filesystem->disk($model->disk)->delete($model->path)) {
-                    $this->db->commit();
-                    return true;
-                }
+                $this->filesystem->disk($model->disk)->delete($model->path);
```

Comment:
We must check the return value and abort/rollback if false

Reply:
Good catch; presumably, nothing in this chain throws an exception?

Reply:
I see your point, now; even though an exception could be thrown on this line, that's not the *only reason* the deletion could fail, e.g., that `delete()` call could return `false` without an exception being thrown. I took a stab at fixing this.